app/controllers/application_controller.rb

```diff
@@ -66,11 +66,13 @@ def current_sp
   end
 
   def sp_from_sp_session
-    ServiceProvider.from_issuer(sp_session[:issuer])
+    sp = ServiceProvider.from_issuer(sp_session[:issuer])
+    sp if sp.is_a? ServiceProvider
```

Comment:
Perhaps, but I think that's for another PR. When I first started working on this part of the code, I questioned whether we need `NullServiceProvider` at all. I'd like to look into this later on. For now, I wanted the simplest way to fix this bug without introducing anything new given our launch deadline.